server/manager/src/main/java/org/apache/accumulo/manager/tableOps/tableImport/PopulateMetadataTable.java

```diff
@@ -176,6 +183,12 @@ public Repo<Manager> call(long tid, Manager manager) throws Exception {
               sawHostingGoal = false;
             }
 
+            if (initialHostingGoal != null) {
+              // add the initial hosting goal
+              HostingColumnFamily.GOAL_COLUMN.put(m,
+                  TabletHostingGoalUtil.toValue(initialHostingGoal));
+              sawHostingGoal = true;
+            }
```

Comment:
should this be an `else` ? Seems that sawHostingGoalUtil may be overwritten with the condition that follows?

Reply:
Not necessarily. It is a bit confusing but this code checks if initialHostingGoal was provided and if so it sets it. Then it checks if something is set in the column. The way the code is now, the initialHostingGoal does not have to be provided so if it is null, the check still needs to happen if there is a hosting goal already set in the column.

Reply:
I think this logic can be cleaned up though now that we will not allow the hosting goal in `ImportConfigurationImpl`. It will make it so that this should never be null.